client/my-sites/stats/activity-log-tasklist/index.jsx

```diff
@@ -0,0 +1,290 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import { isEmpty, get, merge, each, omit, keyBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import ActivityLogTaskUpdate from './update';
+import alPluginsToUpdate from './plugins-to-update';
+import Card from 'components/card';
+import PopoverMenuItem from 'components/popover/menu-item';
+import EllipsisMenu from 'components/ellipsis-menu';
+import Gridicon from 'gridicons';
+import { getSite } from 'state/sites/selectors';
+import { updatePlugin } from 'state/plugins/installed/actions';
+import { getStatusForPlugin } from 'state/plugins/installed/selectors';
+import PluginNotices from 'lib/plugins/notices';
+import { errorNotice, infoNotice, successNotice, removeNotice } from 'state/notices/actions';
+
+/**
+ * Checks if the supplied plugin is currently updating.
+ *
+ * @param {object} plugin Plugin to check status
+ * @returns {bool} True if plugin is updating.
+ */
+const isPluginUpdating = plugin => 'inProgress' === plugin.updateStatus.status;
+
+class ActivityLogTasklist extends Component {
+	static propTypes = {
+		siteId: PropTypes.number,
+		pluginsWithUpdates: PropTypes.arrayOf( PropTypes.object ), // Plugins with pending updates
+		allPluginsToUpdate: PropTypes.arrayOf( PropTypes.object ), // Don't use. Use connected pluginsToUpdate instead.
+
+		// Connected props
+		siteName: PropTypes.string.isRequired,
+		// Plugins already updated + those with pending updates.
+		// This extends allPluginsToUpdate with the plugin update status.
+		pluginsToUpdate: PropTypes.object.isRequired,
+
+		// Localize
+		translate: PropTypes.func.isRequired,
+		updateSinglePlugin: PropTypes.func.isRequired,
+		showErrorNotice: PropTypes.func.isRequired,
+		showInfoNotice: PropTypes.func.isRequired,
+		showSuccessNotice: PropTypes.func.isRequired,
+		removeThisNotice: PropTypes.func.isRequired,
+	};
+
+	state = {
+		pluginUpdateNotice: null, // keyed by plugin id, like "hello-dolly/hello"
+		dismissedPlugins: [],
+		dismissAll: false,
+	};
+
+	dismissAll = () => {
+		// ToDo: this should update some record in the tasklist API
+		this.setState( { dismissAll: true } );
+	};
```

Comment:
note that if we get new plugins after clicking dismiss all they won't appear.
if we want to though we can track which ones were being dismissed…

```js
dismissAll = () => {
	this.setState( {
		// assuming `pluginsToUpdate` is an object whose keys are slugs
		dismissed: union( this.state.dismissed, Object.keys( this.props.pluginsToUpdate ) )
	} );
}
```